pkgs/development/haskell-modules/configuration-ghcjs-9.x.nix

```diff
@@ -19,4 +20,8 @@ with haskellLib;
   ghcjs-dom-javascript = addBuildDepend self.ghcjs-base super.ghcjs-dom-javascript;
   jsaddle = addBuildDepend self.ghcjs-base super.jsaddle;
   jsaddle-dom = addBuildDepend self.ghcjs-base super.jsaddle-dom;
+
+  reflex-dom = overrideCabal (drv: {
+    libraryHaskellDepends = with self; [ reflex reflex-dom-core ];
```

Comment:
If I understand this correctly this is better solved by doing `.override { ... = null; }` for the dependencies you are trying to remove here.